app/src/main/java/bg/zahov/app/data/provider/WorkoutProviderImpl.kt

```diff
@@ -136,6 +144,17 @@ class WorkoutProviderImpl : WorkoutProvider {
     }
 }
 
+/**
+ * Converts a `Workout` object to a `StartWorkout` object.
+ */
+fun Workout.toStartWorkout(): StartWorkout = StartWorkout(
+    id = this.id,
+    name = this.name,
+    date = this.date.toFormattedString(),
+    exercises = this.exercises.map { "${if (it.sets.isNotEmpty()) "${it.sets.size} x " else ""}${it.name} " },
```

Comment:
I suggest using string concateonation ( + operator) here to make the expression easier to read. Also it seems there's empty space at the end of this string.

Reply:
👍

Reply:
👍